### PR TITLE
Support MNVs

### PR DESCRIPTION
**Pull request by kgururaj**
Handle MNVs as described in https://github.com/broadinstitute/gatk/issues/6500

### CODE REVIEW

File: tests/run.py [hunk 1]
```diff
@@ -1399,15 +1412,17 @@ def main():
         if('golden_output' in test_params_dict):
             golden_stdout, golden_md5sum = get_file_content_and_md5sum(test_params_dict['golden_output']);
             if(golden_md5sum != md5sum_hash_str):
+	        is_error = True
                 if(bcftools_path is None):
                     sys.stderr.write('Did not find bcftools in path. If you are able to add that to the path, we can get a more precise diff\n')
```
Comment (kgururaj) [inline tests/run.py:1417]: There was no error raised in this if statement. Tests were reporting error but the script wasn't failing. Can whoever modified this script with extra if-else statements take another look to see if there are similar issues.
See errors in the log of a previous PR:  https://api.travis-ci.org/v3/job/672595164/log.txt
Fortunately, no serious errors went undetected.

File: src/main/cpp/src/utils/logger.cc [hunk 2]
```diff
@@ -43,7 +43,7 @@ Logger logger;
 Logger::Logger() {
   m_logger = spdlog::get(LOGGER_NAME);
   if (m_logger == nullptr) {
-    m_logger = spdlog::stdout_color_mt(LOGGER_NAME);
+    m_logger = spdlog::stderr_color_mt(LOGGER_NAME);
```
Comment (kgururaj) [inline src/main/cpp/src/utils/logger.cc:46]: This was the error that wasn't being caught in previous PRs.

Comment (nalinigans) [reply src/main/cpp/src/utils/logger.cc:46]: My bad 👎 . Thanks for fixing.

**Comment by nalinigans** [comment]
@kgururaj, changes look good to me.
@mlathara, `[travis builds ](https://travis-ci.org/github/GenomicsDB/GenomicsDB/builds/673543901)`are failing. But there is `Did not find bcftools in path. If you are able to add that to the path, we can get a more precise diff` logged. Can you take a look please?

File: tests/run.py (hunk 1 above)
Comment (mlathara) [reply tests/run.py:1417]: Whoops. That was me I think. I'll take a look, and maybe try to add bcftools install to the travis build to see if that helps the travis build.

**Comment by mlathara** [comment]
I guess the travis builds don't install bcftools. I'll look to do that...but also, the Linux build failure isn't strictly a vcf field reordering issue. The test output seems to be missing the ID field entirely. Test output is something like:
```
 1       17386           G       *,<NON_REF>
```
Where we expect something like:
```
1       17386   .       G       *,<NON_REF>
```
Note the missing `.`. Presumably, this happened a bit ago, but we're only seeing it now because of the goof in run.py I added...

File: .travis.yml [hunk 3]
```diff
@@ -129,7 +129,7 @@ script:
...
+        cmake $TRAVIS_BUILD_DIR -DBUILD_JAVA=1 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$GENOMICSDB_INSTALL_DIR -DLIBCSV_DIR=$TRAVIS_BUILD_DIR/dependencies/libcsv -DGENOMICSDB_RELEASE_VERSION=$GENOMICSDB_RELEASE_VERSION -DMAVEN_QUIET=True -DENABLE_LIBCURL=False -DPROTOBUF_LIBRARY=$PROTOBUF_HOME/lib/libprotobuf.dylib -DPROTOBUF_INCLUDE_DIRS=$PROTOBUF_HOME/include -DPROTOBUF_PROTOC_EXECUTABLE=$PROTOBUF_HOME/bin/protoc -DPROTOBUF_PROTOC_LIBRARY=$PROTOBUF_HOME/lib/libprotoc.dylib -DPROTOBUF_LIBRARIES=$PROTOBUF_HOME/lib/libprotobuf.dylib -DOPENSSL_PREFIX_DIR=/usr/local/opt/openssl -DCMAKE_C_COMPILER=/usr/local/opt/mpich/bin/mpicc -DCMAKE_CXX_COMPILER=/usr/local/opt/mpich/bin/mpicxx;
```
Comment (kgururaj) [inline .travis.yml:132]: When compiled in Release mode, unordered_set (hash) gets used in a few places and the order of elements isn't deterministic across systems. In Debug/Coverage mode, set gets used which is ordered. This makes the output byte-wise identical to the golden output. The output is functionally correct in Release mode, just not byte-wise exact.

File: src/main/cpp/src/query_operations/broad_combined_gvcf.cc [hunk 4]
```diff
@@ -796,8 +796,7 @@ void BroadCombinedGVCFOperator::operate(Variant& variant, const VariantQueryConf
   if (m_query_config->is_defined_query_idx_for_known_field_enum(GVCF_ID_IDX)) {
     auto ID_query_idx = m_query_config->get_query_idx_for_known_field_enum(GVCF_ID_IDX);
     merge_ID_field(variant, ID_query_idx);
-    if (!m_ID_value.empty())
-      bcf_update_id(m_vcf_hdr, m_bcf_out, m_ID_value.c_str());
+    bcf_update_id(m_vcf_hdr, m_bcf_out, m_ID_value.empty() ? "." : m_ID_value.c_str());
```
Comment (kgururaj) [inline src/main/cpp/src/query_operations/broad_combined_gvcf.cc:799]: Strange that earlier PRs didn't catch this - again functionally correct, byte-wise different.

File: .travis.yml (hunk 3 above)
Comment (nalinigans) [reply .travis.yml:141]: Yes, not sure why it was compiled in Release mode in the first place. We should make this Coverage too at some point so we get MacOS coverage reported to `codecov`.

File: src/main/cpp/src/query_operations/broad_combined_gvcf.cc (hunk 4 above)
Comment (nalinigans) [reply src/main/cpp/src/query_operations/broad_combined_gvcf.cc:799]: Wonder if some earlier changes have introduced a subtle change in behavior?

File: .travis.yml (hunk 3 above)
Comment (mlathara) [reply .travis.yml:141]: Ah - if we get deterministic order in debug then all that bcftools/vcfdiff stuff could probably be taken out of run.py...

**Pull request merged**